compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/PromptAsCodeCodeModificationTask.java

```diff
@@ -282,75 +317,196 @@ private boolean isNotNPCallCall(FunctionCallExpressionNode functionCallExpressio
         }
     }
 
-    private static class FunctionModifier extends TreeModifier {
+    private static FunctionCallExpressionNode createNPCallFunctionCallExpression(
+            String npPrefix, NaturalExpressionNode naturalExpressionNode, SemanticModel semanticModel) {
+        NodeList<Node> prompt = getRawTemplateContent(naturalExpressionNode, semanticModel);
+        TemplateExpressionNode promptRawTemplate = NodeFactory.createTemplateExpressionNode(
+                SyntaxKind.RAW_TEMPLATE_EXPRESSION, null, BACKTICK, prompt, BACKTICK);
+
+        Optional<ExpressionNode> naturalModelArg = Optional.empty();
+
+        Optional<ParenthesizedArgList> parenthesizedArgListOptional = naturalExpressionNode.parenthesizedArgList();
+        if (parenthesizedArgListOptional.isPresent()) {
+            ParenthesizedArgList parenthesizedArgList = parenthesizedArgListOptional.get();
+            SeparatedNodeList<FunctionArgumentNode> arguments = parenthesizedArgList.arguments();
+            if (!arguments.isEmpty()) {
+                FunctionArgumentNode functionArgumentNode = arguments.get(0);
+                if (functionArgumentNode instanceof PositionalArgumentNode positionalArgumentNode) {
+                    // TODO: validation to be done first
+                    naturalModelArg = Optional.of(positionalArgumentNode.expression());
+                }
+            }
+        }
 
-        private final ModifierData modifierData;
+        SeparatedNodeList<FunctionArgumentNode> arguments =
+                naturalModelArg.isPresent() ?
+                        NodeFactory.createSeparatedNodeList(
+                                promptRawTemplate,
+                                COMMA,
+                                createContextMappingExpressionNode(naturalModelArg.get())
+                        ) :
+                        NodeFactory.createSeparatedNodeList(
+                                promptRawTemplate
+                        );
+        return NodeFactory.createFunctionCallExpressionNode(
+                createNPCallQualifiedNameReferenceNode(npPrefix),
+                OPEN_PAREN,
+                arguments,
```

Comment:
Is this cover the `context` parameter?

Reply:
We expect specifying the model here. Not Context.